### PR TITLE
Workflow / ENH: Add SSH into our runners workflow

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PR adds a new workflow that enables developers to easily SSH into our runners and debug the failing tests, as discussed offline with @ydshieh during lunch

cc @glegendre01

### CODE REVIEW

File: .github/workflows/ssh-runner.yml
Comment (ydshieh) [inline .github/workflows/ssh-runner.yml:51]: why not just use the docker image of daily CI and other steps (except the tests and report part). That way, it's more close to the actual env of the test environment, no?

File: .github/workflows/ssh-runner.yml
Comment (ydshieh) [inline .github/workflows/ssh-runner.yml:59]: @glegendre01 Is it possible to timeout the job with explicit amount of time here?

File: .github/workflows/ssh-runner.yml
Comment (younesbelkada) [reply .github/workflows/ssh-runner.yml]: See: [522ed15](https://github.com/huggingface/transformers/pull/30425/commits/522ed1578b1f1294ad5656ad1a4b3ce426f6c706)

**Review by ydshieh — APPROVED** [review]
LGTM, but let's check how it works after it is merged to `main`.

Thanks a lot for this!

File: .github/workflows/ssh-runner.yml
Comment (glegendre01) [reply .github/workflows/ssh-runner.yml:60]: yes it's possible. 
`sshTimeout: 10m for example`

Comment (ydshieh) [reply .github/workflows/ssh-runner.yml:60]: Hi @glegendre01  I didn't see this `sshTimeout` in [this file](https://github.com/huggingface/tailscale-action/blob/main/action.yaml) - hope I am looking the right file.

What I have in mind (from my experience with CircleCI), there are 2 interesting timeout:

- timeout the job if no ssh connection being done with X minutes (say 10m)
- timeout the job after Y minutes even if there is connection (say 1 hour)

Probably this/these is/are already done in [this file](https://github.com/huggingface/tailscale-action/blob/main/action.yaml) without having them in the `inputs`? (I don't mean to ask them being in `inputs`, just wondering if such timeout are implemented). I do see `timeout 5m` in that file, but not sure its meaning.

Comment (younesbelkada) [reply .github/workflows/ssh-runner.yml:60]: i think tha tfile is different, you need to look for the file with the `v1` tag: https://github.com/huggingface/tailscale-action/blob/v1/action.yaml#L59 🤯 for the other points, i will let @glegendre01 answer 🙏

Comment (ydshieh) [reply .github/workflows/ssh-runner.yml:60]: Nice @younesbelkada thanks a lot

. I think so far we can add `sshTimeout` then merge the PR.

Comment (younesbelkada) [reply .github/workflows/ssh-runner.yml:60]: Thanks ! Ok yes, I think we can use the default one (5min), I would love to have a review from @glegendre01 before merging it if possible 🙏

**Review by glegendre01 — APPROVED** [review]
good idea !

File: .github/workflows/ssh-runner.yml
Comment (ydshieh) [reply .github/workflows/ssh-runner.yml:60]: Oh yes, there is default value 😅

**Pull request merged**
